pgmpy/models/BayesianNetwork.py

```diff
@@ -1364,7 +1363,51 @@ def simulate(
                 model.add_cpds(new_cpd)
                 evidence[new_var] = 0
 
-        # Step 3: If no evidence do a forward sampling
+        # Step 3: If missing_prob; include missing values in samples.
+        if missing_prob:
```

Comment:
I would suggest being more explicit here. Something like `if missing_prob is not None`